### PR TITLE
Ruby 3 compatibility

### PR DESCRIPTION
**Pull request by bquorning**
As you can see, the `alias_method` approach where arguments `(query, *args, &block)` are being explicitly passed around doesn’t work on Ruby 3.0. Is there anything preventing changing the implementation to use `prepend` and `super`?

### CODE REVIEW

**Comment by bquorning** [comment]
cc @grosser

**Comment by grosser** [comment]
prepend+super can work, but sometimes breaks other gems that use alias_method_chain
so need to try this out on a branch and see if it works

**Comment by bquorning** [comment]
> prepend+super can work

It turned out to not be necessary.

**Review by HeyNonster — APPROVED** [review]

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -24,9 +24,9 @@
 
 ActiveRecord::ConnectionAdapters::SQLite3Adapter.class_eval do
   alias_method :exec_query_without_log, :exec_query
-  def exec_query(query, *args, &block)
+  def exec_query(query, *args, prepare: false, &block)
```
Comment (grosser) [inline spec/spec_helper.rb:27]: would prefer if we can do `...` or `**kwargs` ... but this is good enough for now

**Pull request merged**